new doc title and revised intro

joseph-fajen opened the pull request:
<!--
Here are some checklists you may like to use. Use your judgement.

This is just a checklist, all the normative suggestions are covered in more detail in CONTRIBUTING.
-->
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [ ] Reviewer requested

joseph-fajen commented:
This is for PLT-601. I've made these changes: 
1. Changed the project name to "Plutus Tools Developer Guide" 
2. Added a brief introduction of Plutus Tools under a new main heading at the top: "Plutus Tools Developer Guide"
3. Added some info to the section under the heading "The Plutus Platform"

raduom commented on doc/index.rst:
```diff
@@ -1,9 +1,32 @@
 
-The Plutus Platform is an application development platform for the Cardano blockchain.
+Plutus Tools are a collection of off-chain infrastructure resources. Plutus Tools are built 
+for developers who are authoring web-based applications that implement smart contracts on the 
```
```suggestion
for developers who are authoring applications that implement smart contracts on the 
```

raduom replied:
There is nothing web-specific about the tools.

joseph-fajen replied:
Great to know, thanks!

joseph-fajen replied:
I'm thinking we could say "for developers who are authoring distributed applications that implement smart contracts on the ..." Or are they not necessarily always going to be distributed applications?

raduom commented on doc/index.rst:
```diff
@@ -1,9 +1,32 @@
+
+* Libraries which implement the Plutus Application Framework
+* A selection of end-to-end use cases written with the Plutus Application Framework
+* The Plutus Playground, a web-based playground for learning and writing basic Plutus Applications
```
Didn't we decide we don't want to support this?

joseph-fajen replied:
I got these points from https://github.com/input-output-hk/plutus-apps#readme. Maybe this info is out of date. Are you referring to the Plutus Playground, or all 3 points here?

raduom replied:
@asutherlandus are we still supporting the playground?

joseph-fajen replied:
I'm removing this reference to the Plutus Playground.

raduom commented on doc/index.rst:
```diff
@@ -1,9 +1,32 @@
+the entire tool box where you can find a collection of utilities and components. Pick and choose 
+from the tools and examples that best suit your use case. 
+
+The Plutus Platform
```
I am not sure about the formulation of this section. It sounds like we have a platform, but I don't think that we really do.

For example I do not know what the Plutus Application Framework consists of, I am not sure we have any end-to-end use cases, etc.

joseph-fajen replied:
Thanks for your feedback. Do you think the info on this page is still largely accurate: 
https://plutus-apps.readthedocs.io/en/latest/plutus/explanations/platform.html

I think it may be important and useful to have a conversation about how we describe and characterize what Plutus Tools is. I have found this to be a challenging subject to have clarity about. 

I'd also like to learn more about the meaning of the content on these pages: 

https://plutus-apps.readthedocs.io/en/latest/reference/examples.html

https://github.com/input-output-hk/plutus-apps/blob/main/plutus-use-cases/ARCHITECTURE.adoc

michaelpj replied:
no, the stuff about the plutus platform is absolutely not accurate

raduom reviewed (commented):
I am not sure if this is a future plan or description of what we currently have.

asutherlandus commented on doc/index.rst:
```diff
@@ -1,9 +1,32 @@
 
-The Plutus Platform is an application development platform for the Cardano blockchain.
+Plutus Tools are a collection of off-chain infrastructure resources. Plutus Tools are built 
+for developers who are authoring web-based applications that implement smart contracts on the 
```
I think cloud-based may be a more accurate term.
https://www.techtarget.com/searchcloudcomputing/definition/cloud-application#:~:text=A%20key%20difference%20between%20cloud,a%20local%20computer%20or%20workstation.

asutherlandus commented on doc/index.rst:
```diff
@@ -1,9 +1,32 @@
+for developers who are authoring web-based applications that implement smart contracts on the 
+Cardano blockchain. 
+
+The Plutus Tools Team supports the underlying resources that developers need who are 
```
Is it better to refer to the contents of the repo rather than the team?

joseph-fajen replied:
Yes, probably so. "The Plutus Tools repository supports the underlying resources..."

asutherlandus commented on doc/index.rst:
```diff
@@ -1,9 +1,32 @@
+The Plutus Tools Team supports the underlying resources that developers need who are 
+writing full applications using Plutus in Haskell, including off-chain code. 
+The term “off-chain code” refers to the part of a contract application’s code which 
+runs off the chain, usually as a contract application. 
```
redundant use of contract application

joseph-fajen replied:
True, did some pruning.

asutherlandus commented on doc/index.rst:
```diff
@@ -1,9 +1,32 @@
+
+* Libraries which implement the Plutus Application Framework
+* A selection of end-to-end use cases written with the Plutus Application Framework
+* The Plutus Playground, a web-based playground for learning and writing basic Plutus Applications
```
We should mark Plutus playground deprecated

joseph-fajen replied:
I'll remove this reference to Plutus Playground. We'll also need to update the readme for https://github.com/input-output-hk/plutus-apps.

michaelpj commented on doc/index.rst:
```diff
@@ -1,9 +1,32 @@
+
+The Plutus Tools Team supports the underlying resources that developers need who are 
+writing full applications using Plutus in Haskell, including off-chain code. 
+The term “off-chain code” refers to the part of a contract application’s code which 
```
Should this just link to the glossary?

joseph-fajen replied:
Good point. However, I want the reader to have a sufficient general understanding of this concept without needing to leave the landing page.

joseph-fajen commented on doc/index.rst:
```diff
@@ -1,9 +1,32 @@
-The Plutus Platform
+Plutus Tools Developer Guide
```
After a conversation with Konstantinos and Andrew, we're trying on "Plutus Tools SDK User Guide."

joseph-fajen commented on doc/index.rst:
```diff
@@ -1,9 +1,32 @@
 ===============================
 
-The Plutus Platform is an application development platform for the Cardano blockchain.
+Plutus Tools are a collection of off-chain infrastructure resources. Plutus Tools are built 
```
Good point. My first impulse was to use a singular reference, but then I thought about it too much and went with plural!

joseph-fajen commented:
I'm having some trouble with trying to push changes for my 2nd commit on this PR. I've reached out to Lorenzo. My intention is to update this PR for another round of reviews before merging with main. I'm seeing a message above that I've asked to merge two commits into main. I think I probably used an incorrect git command.

joseph-fajen commented:
The changes in the most recent push on Aug. 14 were: 

1. Changed name of doc to "Plutus Tools SDK User Guide"
2. Updated brief intro on landing page to call out intended audience
3. Added a "note" for readers who want to contribute to Plutus Tools SDK to refer to the documentation in the repo
4. Added links to a few items as examples of Plutus Tools packages. I expect that these links will need to be added to and further edited, but this reflects is what I'm aware of at this point. 
5. Pointing readers to https://github.com/input-output-hk/plutus-starter as a place to reference for getting started.

koslambrou commented on doc/index.rst:
```diff
@@ -1,10 +1,44 @@
-The Plutus Platform is an application development platform for the Cardano blockchain.
-To read more about the Platform, see :ref:`what_is_the_plutus_platform`.
-To get started using the Platform see :ref:`plutus_getting_started`.
+Intended Audience
+-------------------------------
 
+External Developers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Isn't it redundant to have a separate section for "External developers" when this doc is specifically for external developers?

michaelpj replied:
Also, this is pretty weird to use in an users guide! From the users guide perspective "normal" developers are "external", we're weird for being "internal"!

joseph-fajen replied:
We could probably go without the "External Developers" heading. My goal is that if a reader who knew nothing about any of this came to this page for the first time, I want them to understand what this documentation is, who it's for, why they may or may not care about it. Easy to understand and clear.

koslambrou commented on doc/index.rst:
```diff
@@ -1,10 +1,44 @@
+input-output-hk/plutus-apps>`_ supports the underlying resources that developers 
+need who are writing full applications using Plutus in Haskell, including off-chain 
+code. The term “off-chain code” refers to the part of a contract application’s code 
+which runs off the chain. 
```
For more precision, how about "... which runs outside the blockchain"?

joseph-fajen replied:
Yes, I think that's a helpful description.

koslambrou commented on doc/index.rst:
```diff
@@ -1,10 +1,44 @@
+
+* a variety of other Plutus packages. 
+
+To get started using the Plutus Tools SDK, see the `Plutus starter template repository 
```
I suggest: "See the `Plutus starter template repository <https://github.com/input-output-hk/plutus-starter>`_ for a simple starter project using the Plutus Tools SDK.

joseph-fajen replied:
Suggestion incorporated.

koslambrou reviewed (approved):
Very nice :) Let's see what others think.

raduom commented on doc/index.rst:
```diff
@@ -1,10 +1,44 @@
+for external developers. The `Plutus Tools SDK repository <https://github.com/
+input-output-hk/plutus-apps>`_ supports the underlying resources that developers 
+need who are writing full applications using Plutus in Haskell, including off-chain 
+code. The term “off-chain code” refers to the part of a contract application’s code 
+which runs off the chain. 
```
```suggestion
code. The term “off-chain code” refers to the part of a contract application’s code 
which responds to events happening on the blockchain, usually by producing transaction. 
```
I am not sure about the correct, but the original text sounds a bit like a circular definition.

joseph-fajen replied:
This adds some useful perspective on what off-chain code is doing.

joseph-fajen replied:
@koslambrou @raduom In light of your suggestions, this is what I'm going with right now: 

> The term “off-chain code” refers to the part of a contract application’s code 
> which runs outside of the blockchain. Off-chain code responds to events happening 
> on the blockchain, usually by producing transactions.

raduom commented on doc/index.rst:
```diff
@@ -1,10 +1,44 @@
+  a lightweight, customizable chain follower application and library for DApp 
+  developers who need to index and query the Cardano blockchain; 
+
+* a variety of other Plutus packages. 
```
I think this says too little. Some things that I would like to know:
* What packages? Are they packages or Haskell modules?
* What are they used for?

I think just 1-2 examples would do, but without those this seems quite superfluous as it does not provide any new information.

joseph-fajen replied:
@raduom I agree with your comment. I would like to include more substantial information here, but I'm unclear about what is available today from Tools that is ready for developers to use. What else could we add here, and what else might we say about what is available? 

@michaelpj Excellent point. I would like to have a basic list of solid Tools offerings here in order to give developers a better sense of what Tools is. I'm ok with updating the list regularly if needed to keep it current. Alternatively, we could link from here to another page that has a more thorough list of available packages. I've talked with Konstantinos about such a list. Related tickets are PLT-444 and PLT-644. 

Are there 2 or 3 more examples we could add here? From the list of a couple of dozen folders on the plutus-apps repo, which ones, if any, do you think would be helpful to reference specifically in this intro? 

This page seems interesting. Would it make sense to mention some of what it mentions? 

https://github.com/input-output-hk/plutus-apps/blob/main/plutus-use-cases/ARCHITECTURE.adoc

raduom replied:
I was suggesting you remove that line. Currently we have: PAB and chain-index. Soon we will also have Marconi, and the PAB and chain-index versions based on Marconi.

raduom replied:
I think that listing the projects in plutus-tools with a brief explanation is a good idea, here.

raduom reviewed (approved):
LGTM

michaelpj commented on doc/index.rst:
```diff
@@ -1,10 +1,44 @@
-The Plutus Platform
+Plutus Tools SDK User Guide
```
Hmm, do we need to duplicate the title here? Can we get sphinx to print it here or something?

joseph-fajen replied:
Good question. When I removed the H1 heading here, it also impacted the title displaying above. I would think there would be a way to avoid the duplication. I'll look into that.

joseph-fajen replied:
In the current versions of both RTD instances, the title is duplicated. I would like to find a way to not duplicate it, but just want to note that this isn't a new issue. But I'll look into this to see if we can improve this behavior.

michaelpj commented on doc/index.rst:
```diff
@@ -1,10 +1,44 @@
+External Developers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This user guide is intended for external developers who are authoring distributed 
```
```suggestion
This user guide is intended for developers who are authoring distributed 
```

joseph-fajen replied:
Suggestion incorporated. 👍

joseph-fajen commented on doc/index.rst:
```diff
@@ -1,10 +1,44 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This user guide is intended for external developers who are authoring distributed 
+applications ("DApps") that implement smart contracts on the Cardano blockchain. 
```
Suggestion incorporated. 👍

michaelpj commented:
@joseph-fajen you've done something odd with git and pulled in some commits you probably shouldn't have

james-iohk commented on doc/index.rst:
```diff
@@ -1,10 +1,55 @@
+   please refer to documentation residing in the `Plutus Tools SDK repository <https://github.com/
+   input-output-hk/plutus-apps>`_.
+
+Plutus Tools SDK
```
I think the Trace Emulator is worth highlighting here as a tool. It's used for testing plutus contracts on an emulated blockchain. https://github.com/input-output-hk/plutus-apps/blob/main/plutus-contract/src/Plutus/Trace/Emulator.hs

joseph-fajen commented:
@michaelpj Yes, I've done something odd for sure! I'm not sure why my PR includes those 3 ADR files. I've asked Lorenzo to help me sort this out. Otherwise, I'm thinking of my most recent commit as a viable version to merge unless new comments arise.

raduom reviewed (commented):
LGTM, but you need to figure out the two new ADRs :)

eyeinsky commented on doc/index.rst:
```diff
@@ -1,10 +1,71 @@
+input-output-hk/plutus-apps>`_ supports the underlying resources that developers 
+need who are writing full applications using Plutus in Haskell, including off-chain 
+code. The term “off-chain code” refers to the part of a contract application’s code 
+which runs outside of the blockchain. Off-chain code responds to events happening 
```
Not entirely sure but wasn't it the *on-chain* code that responded to events from off-chain? At least that's what I vaguely remember from the Plutus Pioneer Program.

james-iohk replied:
The onchain checks whether certain conditions are met with limited scope. The only "response" it can perform is to invalidate a transaction by returning an error. Off-chain code can respond to any event happening, including on this blockchain or from any other interface.
